src/pages/WizkidEdit.js

```diff
@@ -30,30 +31,47 @@ function WizkidEdit() {
     }
   };
 
+  const handleSubmitChangeEmployement= async () => {
+    try {
+      await setEmployementWizkidById(id);
+      navigate("/");
+    } catch(error) {
+      setError(error.message);
+    }
+  };
+
 const found = wizkids.find(element => element.id === Number(id));
 useEffect(() => {
   if(found !== undefined) {
     setName(found.name);
+    setEmployed(found.employed);
   } 
 }, [found]);
 
 if(loading) {
   return <LoadingContainer />
 }
 
+const employmentButtonText = employed ? "Fire!" : "Hire back!";
+
   return (
     <div>
       <Navbar />
       {found !== undefined && (
         <WizkidElementContainer>
-        <form className="wizkid-edit-form" onSubmit={handleSubmit}>
+        <form className="wizkid-edit-form" onSubmit={handleSubmitEditName}>
             <label className="label-form">Name</label>
             <input value={name} className="input-form" onChange={handleChange}/>
             <button className="wizkid-edit-button-edit">
                 Save
             </button>
             {error && <p style={{color: "red", fontWeight: "bold"}}>{error}</p>}
           </form>
+          {token && (
+            <button className="wizkid-edit-button-set-employement" onClick={handleSubmitChangeEmployement}>
+              {employmentButtonText}
+            </button>
+          )}
```

Comment:
Maybe a better solution is to not even allow a "guest"  on the page at all by adding a check on the router level

Reply:
You mean something like this?
`  return (
    <div>
      <Routes>
        <Route path="/" element={<Home />} />
        <Route path="/details/:id" element={<WizkidDetails />} />
        {token && <Route path="/edit/:id" element={<WizkidEdit />} />}
        <Route path="/login" element={<Login />} />
      </Routes>
    </div>

  );`